hera_cal/chunker.py

```diff
@@ -0,0 +1,144 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+# Copyright 2021 the HERA Project
+# Licensed under the MIT License
+
+
+from . import io
+import argparse
+import numpy as np
+import sys
+import warnings
+
+
+def chunk_data_files(filenames, inputfile, outputfile, chunk_size,
+                     polarizations=None, spw_range=None, throw_away_flagged_bls=False,
+                     clobber=False, ant_flag_yaml=None):
+    """A data file chunker
+
+    Parameters
+    ----------
+    filenames: list of strings
+        list of filenames to chunk. Should be homogenous in blt.
+    outpufile: str
+        name of outputfile to write time-concatenated data too.
+    polarizations: list of strs, optional
+        Limit output to polarizations listed.
+        Default None selects all polarizations.
+    spw_range: 2-list or 2-tuple
+        optional lower and upper channel range to select
+    throw_away_flagged_bls: bool, optional
+        if true, throw away baselines that are fully flagged.
+        default is False.
+    clobber: bool, optional
+        if true, overwrite any preexisting output files.
+        defualt is false.
+    flag_yaml : str, optional
+        yaml file with list of antennas to flag and throw away if throw_away_flagged_bls is True
+    Returns
+    -------
+    Concatenated HERAData object.
+
+    """
+    filenames = sorted(filenames)
+    start = filenames.index(inputfile)
+    end = start + chunk_size
+    hd = io.HERAData(filenames[start:end])
+    read_args = {}
+    if polarizations is not None:
+        read_args['polarizations'] = polarizations
+    if spw_range is not None:
+        read_args['freq_chans'] = np.arange(spw_range[0], spw_range[1]).astype(int)
+    data, flags, nsamples = hd.read(axis='blt', **read_args)
+    # throw away fully flagged baselines.
+    if throw_away_flagged_bls:
+        from hera_qm.utils import apply_yaml_flags
+        hd = apply_yaml_flags(hd, ant_flag_yaml, flag_freqs=False, flag_times=False,
+                              flag_ants=True, ant_indices_only=True, throw_away_flagged_ants=True)
+
+    hd.write_uvh5(outputfile, clobber=clobber)
+
+    return hd
+
+
+def chunk_cal_files(filenames, inputfile, outputfile, chunk_size, spw_range=None, clobber=False):
+    """A calibration file chunker
```

Comment:
The one-line summary should be a command.

Reply:
Turned this into a command.